src/runtime_src/core/pcie/windows/device_windows.cpp

```diff
@@ -911,6 +929,58 @@ struct data_retention
 
 };
 
+struct mailbox
+{
+    using result_type = boost::any;
+
+    static xcl_mailbox
+        init_mailbox_info(const xrt_core::device* dev)
+    {
+        xcl_mailbox info = { 0 };
+        userpf::get_mailbox_info(dev->get_user_handle(), &info);
+        return info;
+    }
+
+    static result_type
+        get_info(const xrt_core::device* device, key_type key)
+    {
+        static std::map<const xrt_core::device*, xcl_mailbox> info_map;
+        static std::mutex mutex;
+        std::lock_guard<std::mutex> lk(mutex);
+
+        auto it = info_map.find(device);
+        if (it == info_map.end()) {
+            auto ret = info_map.emplace(device, init_mailbox_info(device));
```

Comment:
`ret` is useless, specially with such a bad naming.
```suggestion
            it = info_map.emplace(device, init_mailbox_info(device)).first;
```